### ISSUE TITLE
Move from mailboxprocessor to standard js events

### ISSUE THREAD

**Issue by mastoj** [fable-compiler/fable-arch#41]
The mailboxprocessor is nice, but performance wise I think we could do much better with events. My thoughts are roughly this:

``` js
var elem = document.createElement("div"); // elem to hold state, will never be rendered
var handleEvent = function(e, post) {
    var data = e.details.data;
    var state = doSomeMagic(e.details, elem["state"]);
    elem["state"] = state;
}
var post = function(data) {
    var event  = new CustomEvent("ArchEvent", {details:{data: data}});
    elem.dispatchEvent(event);
}
elem.addEventListener("ArchEvent", function(e) { handleEvent(e,post)});
```

Here we create a element that will hold the state for things in the application, basically the same thing we use when we do the recursive call. We create a helper function for posting messages to that will be used to post messages on the element we created. That we just attach a event listener to the element and celebrates victory.

**Comment by MangelMaxime** [comment]
I would be interesting if we could give it a try.

Did you take a look at how others libraries are doing the Event parts ?

**Comment by mastoj** [comment]
didn't find an easy part that shows how they do it, but didn't look that much either to be honest :)

**Comment by mastoj** [comment]
Solved by #42
